### DISCUSSION TITLE
401 error with File uploader in production environment

### DISCUSSION THREAD

**Discussion by msalehipro** [filamentphp/filament#9243] [Help] [answered]
I'm getting 401 error when i upload a file with default file uploader, in production environment only.
i use laravel 10 and filament 3 and octane with swoole .. also in production we are using nginx and cloudflare (reverse proxy)
i have no problem in local env
i also searched many times about this problem.
i did set the `$proxies` in `TrustProxies` middleware to `'*'`
 and set this in my boot() method in AppServiceProvider :
```php
if($this->app->environment('production')) {
            URL::forceScheme('https');
}
```
Laravel v10.28.0 (PHP v8.2.7)
anyone know what is the reason ?

**Comment by msalehipro** [comment, marked as answer]
my problem solved.
actually it was because of my proxy pass settings in nginx that solved using this one line of code :
`proxy_set_header X-Forwarded-Proto https;`

source: 
https://github.com/livewire/livewire/discussions/3084

**Reply by bkremenovic** [reply]
Thank you so much for this. Been struggling with this for 5 days. Finally solved!
